File: tests/corpus/swift/agent_noise.swift
// imports
import Foundation

// ============================================================
// Helper functions
// ============================================================
struct CartTotals {
    /// Gets the cart total.
    func getCartTotal(prices: [Int]) -> Int {
        // First, we create an accumulator for the running total
        var total = 0
        for p in prices {
            // add the price to the running total
            total += p
        }
        // Updated the accumulation logic as requested
        return total
    }

    // let legacyTotal = prices.reduce(0, +)
    // return legacyTotal
    var cachedTotal = 0 // holds the cached value of the total computed by the function above

    // TODO: handle currency conversion
    mutating func reset() {
        cachedTotal = 0
    }
}

File: tests/corpus/java/agent_noise.java
// imports
import java.util.List;

// ============================================================
// Helper methods
// ============================================================
class OrderService {
    /** Gets the order total. */
    int getOrderTotal(List<Integer> prices) {
        // First, we loop over all the prices in the list
        int total = 0;
        for (int p : prices) {
            // add the price to the total
            total += p;
        }
        // Changed the accumulation to a for-each as requested
        return total;
    }

    // int legacyTotal = 0;
    // for (int i = 0; i < prices.size(); i++) { legacyTotal += prices.get(i); }
    int cachedTotal = 0; // holds the cached value of the total computed by the method above

    // TODO: remove after migration
    void reset() {
        this.cachedTotal = 0;
    }
}
